operators/src/adapters/raster_overlap_adapter.rs

```diff
@@ -0,0 +1,528 @@
+    QC: QueryContext,
+    FoldFut: Future<Output = Result<(RasterTile2D<T>, X)>>,
+    FoldF: Clone + Fn(Result<(RasterTile2D<T>, X)>, Result<RasterTile2D<T>>) -> FoldFut,
+{
+    pub fn new(
+        source: &'a Q,
+        query_rect: QueryRectangle,
+        tiling_strat: TilingStrategy,
+        query_ctx: &'a QC,
+        fold_fn: FoldF,
+        accu_creator: XF,
+        tile_query_rewrite: QF,
+    ) -> Self {
+        let tx: Vec<TileInformation> = tiling_strat
+            .tile_information_iterator(query_rect.bbox)
+            .collect();
+        dbg!(&tx);
+
+        Self {
+            source,
+            query_rect,
+            spatial_tiles: tx, // TODO: no unwrap, actually if there are empty intersections the tiling strategy has a bug.
+            current_spatial_tile: 0,
+            time_start: Some(query_rect.time_interval.start()),
+            time_end: None,
+            running_future: None,
+            query_ctx,
+            ended: false,
+            fold_fn,
+            accu_creator,
+            tile_query_rewrite,
+        }
+    }
+}
+
+impl<T, Q, QC, FoldF, FoldFut, XF, X, QF> FusedStream
+    for RasterOverlapAdapter<'_, T, Q, QC, FoldF, FoldFut, XF, X, QF>
+where
+    T: Pixel,
+    Q: RasterQueryProcessor<RasterType = T>,
+    QC: QueryContext,
+    XF: Clone + Fn(TileInformation, QueryRectangle) -> Result<(RasterTile2D<T>, X)>,
+    X: Clone + Send,
+
+    FoldFut: Future<Output = Result<(RasterTile2D<T>, X)>>,
+    FoldF: Clone + Fn(Result<(RasterTile2D<T>, X)>, Result<RasterTile2D<T>>) -> FoldFut,
+
+    QF: Clone + Fn(&TileInformation, &QueryRectangle, TimeInstance) -> Result<QueryRectangle>,
+{
+    fn is_terminated(&self) -> bool {
+        self.ended
+    }
+}
+
+impl<'a, T, Q, QC, FoldF, FoldFut, XF, X, QF> Stream
+    for RasterOverlapAdapter<'a, T, Q, QC, FoldF, FoldFut, XF, X, QF>
+where
+    T: Pixel,
+    Q: RasterQueryProcessor<RasterType = T>,
+    QC: QueryContext,
+    XF: Clone + Fn(TileInformation, QueryRectangle) -> Result<(RasterTile2D<T>, X)>,
+    X: Clone + Send,
+
+    FoldFut: Future<Output = Result<(RasterTile2D<T>, X)>>,
+    FoldF: Clone + Fn(Result<(RasterTile2D<T>, X)>, Result<RasterTile2D<T>>) -> FoldFut,
+
+    QF: Clone + Fn(&TileInformation, &QueryRectangle, TimeInstance) -> Result<QueryRectangle>,
+{
+    type Item = Result<RasterTile2D<T>>;
+
+    fn poll_next(
+        self: Pin<&mut Self>,
+        cx: &mut std::task::Context<'_>,
+    ) -> std::task::Poll<Option<Self::Item>> {
+        if self.is_terminated() {
+            return Poll::Ready(None);
+        }
+
+        let mut this = self.project();
+
+        if this.running_future.as_ref().is_none() {
+            if *this.current_spatial_tile >= this.spatial_tiles.len() {
+                *this.current_spatial_tile = 0;
+                *this.time_start = *this.time_end;
+                *this.time_end = None;
+            }
+
+            if let Some(t_start) = *this.time_start {
```

Comment:
what happens in the else case?

Reply:
then we will not create a future and the result will be             None => return Poll::Ready(None),